### PR TITLE
feat: add mtls support with user provided cert/key

### PR DESCRIPTION
**Pull request by arithmetic1728**
Add mtls support in gapic-showcase server so all microgenerators can use it for mtls system test in the future. Currently only python microgenerator implemented mtls. Verified with the following python microgenerator generated showcase client:
https://github.com/arithmetic1728/showcase_mtls_sample

### CODE REVIEW

File: cmd/gapic-showcase/run.go [hunk 1]
```diff
@@ -60,6 +67,31 @@ func init() {
 				grpc.StreamInterceptor(observerRegistry.StreamInterceptor),
 				grpc.UnaryInterceptor(observerRegistry.UnaryInterceptor),
 			}
+
+			// load mutual TLS cert/key and root CA cert
+			if tlsCaCert != "" || tlsCert != "" || tlsKey != "" {
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:72]: Should this be `tlsCaCert != "" && tlsCert != "" && tlsKey != ""`? These variables all seem to be required in order to assemble the certificates.

File: cmd/gapic-showcase/run.go [hunk 2]
```diff
@@ -60,6 +67,31 @@ func init() {
...
+		    		x509KeyPair, err := tls.LoadX509KeyPair(tlsCert, tlsKey)
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:73]: Please rename `x509KeyPair`  to just `keyPair`

File: cmd/gapic-showcase/run.go [hunk 3]
```diff
@@ -60,6 +67,31 @@ func init() {
...
+		                if err != nil {
+	            	        	log.Fatalf("Failed to load server TLS cert/key with error:%v", err)
+		    		}
+		    		caCert, err := ioutil.ReadFile(tlsCaCert)
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:77]: Please rename `caCert` to just `cert`

File: cmd/gapic-showcase/run.go [hunk 4]
```diff
@@ -60,6 +67,31 @@ func init() {
...
+		    		if err != nil {
+	            			log.Fatalf("Failed to load root CA cert file with error:%v", err)
+		    		}
+				caCertPool := x509.NewCertPool()
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:81]: Please rename `caCertPool` to just `pool`

File: cmd/gapic-showcase/run.go [hunk 5]
```diff
@@ -60,6 +67,31 @@ func init() {
...
+			// load mutual TLS cert/key and root CA cert
+			if tlsCaCert != "" || tlsCert != "" || tlsKey != "" {
+		    		x509KeyPair, err := tls.LoadX509KeyPair(tlsCert, tlsKey)
+		                if err != nil {
+	            	        	log.Fatalf("Failed to load server TLS cert/key with error:%v", err)
+		    		}
+		    		caCert, err := ioutil.ReadFile(tlsCaCert)
+		    		if err != nil {
+	            			log.Fatalf("Failed to load root CA cert file with error:%v", err)
+		    		}
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:80]:
```suggestion
		    		}

```

File: cmd/gapic-showcase/run.go [hunk 6]
```diff
@@ -60,6 +67,31 @@ func init() {
 				grpc.StreamInterceptor(observerRegistry.StreamInterceptor),
 				grpc.UnaryInterceptor(observerRegistry.UnaryInterceptor),
 			}
+
+			// load mutual TLS cert/key and root CA cert
+			if tlsCaCert != "" || tlsCert != "" || tlsKey != "" {
+		    		x509KeyPair, err := tls.LoadX509KeyPair(tlsCert, tlsKey)
+		                if err != nil {
+	            	        	log.Fatalf("Failed to load server TLS cert/key with error:%v", err)
+		    		}
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:76]:
```suggestion
		    		}

```

File: cmd/gapic-showcase/run.go [hunk 7]
```diff
@@ -60,6 +67,31 @@ func init() {
...
+		    		caCert, err := ioutil.ReadFile(tlsCaCert)
+		    		if err != nil {
+	            			log.Fatalf("Failed to load root CA cert file with error:%v", err)
+		    		}
+				caCertPool := x509.NewCertPool()
+				caCertPool.AppendCertsFromPEM(caCert)
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:82]:
```suggestion
				caCertPool.AppendCertsFromPEM(caCert)

```

File: cmd/gapic-showcase/run.go [hunk 8]
```diff
@@ -60,6 +67,31 @@ func init() {
...
+					Certificates: []tls.Certificate{x509KeyPair},
+					ClientCAs:    caCertPool,
+					ClientAuth:   tls.RequireAndVerifyClientCert,
+				})
+
+				opts = []grpc.ServerOption{
+					grpc.StreamInterceptor(observerRegistry.StreamInterceptor),
+					grpc.UnaryInterceptor(observerRegistry.UnaryInterceptor),
+					grpc.Creds(ta),
+				}
```
Comment (noahdietz) [inline cmd/gapic-showcase/run.go:93]:
```suggestion
				opts = append(opts, grpc.Creds(ta))
```

This slice is already declared and initialized with the interceptors prior to this block, so we don't need to initialize it again.

**Comment by noahdietz** [comment]
BTW the build is failing because of different issue. Breaking change in a dependency I have to handle. I'll let you know how to resolve it once I do.

File: cmd/gapic-showcase/run.go (hunk 1 above)
Comment (arithmetic1728) [reply cmd/gapic-showcase/run.go]: yes, updated the code.

File: cmd/gapic-showcase/run.go (hunk 2 above)
Comment (arithmetic1728) [reply cmd/gapic-showcase/run.go]: done

File: cmd/gapic-showcase/run.go (hunk 3 above)
Comment (arithmetic1728) [reply cmd/gapic-showcase/run.go]: done

File: cmd/gapic-showcase/run.go (hunk 4 above)
Comment (arithmetic1728) [reply cmd/gapic-showcase/run.go]: done

**Comment by arithmetic1728** [comment]
> BTW the build is failing because of different issue. Breaking change in a dependency I have to handle. I'll let you know how to resolve it once I do.

Sure. Thank you Noah!

**Comment by noahdietz** [comment]
I released the tool that was afflicted by the breaking change with the fix and reran the presubmits here so that it pulled the latest version of that tool. It passes and you don't need to do anything!

**Pull request merged**